astropy/uncertainty/core.py

```diff
@@ -37,39 +38,36 @@ class Distribution:
         sole dimension is used as the sampling axis (i.e., it is a scalar
         distribution).
     """
-    _generated_subclasses = {}
-
-    def __new__(cls, samples):
-        if isinstance(samples, Distribution):
-            samples = samples.distribution
+    @classmethod
+    def _get_fallback_generated_subclass(cls):
+        return ArrayDistribution
+
+    @classmethod
+    def _make_generated_subclass(cls, data_cls, base_cls):
+        # TODO: try to deal with this at the lower level.  The problem is
```

Comment:
Needs a bit more of the comment below copied, so it is clear the `TODO` is about trying to implement `__repr__` and `__str__` better. Aside: now fairly certain this will take defining `__array_function__`

Reply:
👍.
There are a lot of docstring that still need doing.
You wrote Masked (and Uncertainty?), so any tips / direct commits are appreciated 😆

Reply:
ping @mhvk for a docstring review.